Sources/ToucanSource/Models/PipelineResult.swift

```diff
@@ -7,24 +7,49 @@
 
 /// Represents the output of a content transformation pipeline, including the
 /// transformed content and its intended destination.
-public struct PipelineResult {
+public struct PipelineResult: Sendable {
 
-    /// The final transformed content (e.g., HTML, Markdown, etc.).
-    public var contents: String
+    /// The source material for the pipeline result.
+    public enum Source: Sendable {
+        /// The original source material, source path and optional Contents
+        case asset(String, String?)
+        /// The final transformed content (e.g., HTML, Markdown, etc.).
+        case content(String)
+    }
+
+    /// The source material.
+    public var source: Source
 
     /// The destination metadata describing where or how the content should be output.
     public var destination: Destination
 
     /// Initializes a new `PipelineResult` with transformed content and a destination.
     ///
     /// - Parameters:
-    ///   - contents: The transformed content string.
+    ///   - source: The source material.
     ///   - destination: A `Destination` indicating where the result should be saved or rendered.
     public init(
-        contents: String,
+        source: Source,
         destination: Destination
     ) {
-        self.contents = contents
+        self.source = source
         self.destination = destination
     }
+
+    /// A Boolean value indicating whether the pipeline result's source is content-based.
+    /// Returns `true` if the source is `.content`, otherwise `false`.
+    public var isContent: Bool {
```

Comment:
I don't think `isContent` and `isAsset` is necessary if we use proper cases.

Reply:
I'm using this when filtering results, it's handy, but I'll move the extension to the source type.